### DISCUSSION TITLE
Release Long Term Support (LTS) version

### DISCUSSION THREAD

**Discussion by y-tabata** [keycloak/keycloak#25688] [Ideas]
Currently, Keycloak is planned for four major version releases per year [1].
When considering applying Keycloak to a production environment, countermeasures against vulnerabilities are one of the most important things. Still, while some projects can follow the major version updates of Keycloak at the above frequency, others cannot. In particular, in large-scale or high-security environments, that require numerous testing, frequent version updates may not be possible.
By releasing the LTS version, Keycloak has the potential to be used in even more situations.
In this discussion, we hope to gather and discuss the actual demand for LTS and expectations for LTS.
WDYT?

[1] https://github.com/keycloak/keycloak/milestones

**Comment by sschu** [comment]
There is an LTS-Version of Keycloak which is RHBK (Red Hat build of Keycloak).

**Reply by y-tabata** [reply]
RHBK is a commercial version. I want to discuss the LTS of the community version here.

**Reply by amiv1** [reply]
I'd say RHBK is also not LTS, the maintenance support of a major version ends in 1 year according to [product lifecycles page](https://access.redhat.com/product-life-cycles?product=Red%20Hat%20build%20of%20Keycloak).
Full support ends even faster - in 6 months.

This is pretty much a rolling release.

**Comment by yogee246** [comment]
Was surprised to see no LTS version available for community version. Any updates on the LTS release or atleast what has worked best among the latest releases for a larger community?

**Comment by jonkoops** [comment]
Keycloak itself does not have LTS releases, nor do we intend to create such releases at this point in time. If you are looking for extended support then there is the [Red Hat build of Keycloak](https://access.redhat.com/products/red-hat-build-of-keycloak/). We are planning on doing some versioning changes starting with Keycloak 26 that will be relevant to this issue, can't say anything else about this at this time, but keep an eye on our blog posts.
